gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2411,4 +2438,25 @@ private boolean requesterShouldPay(String bucketName) {
   public AtomicLong getObjectStatistics(GoogleCloudStorageStatistics key) {
     return objectStatistics.get(key);
   }
+
+  @Override
+  public AtomicLong getHttpStatistics(GoogleCloudStorageStatistics key) {
+    setHttpStatistics();
```

Comment:
Looks like everytime we get a statistic, we end up updating the entire dataset, which is inefficient. Please refactor